demo/full/scripts/components/BufferContentGraph.jsx

```diff
@@ -0,0 +1,251 @@
+      const endPoint = Math.min(end - minimumPosition, maximumPosition);
+      const scaledStart = startPoint / wholeDuration;
+      const scaledEnd = endPoint / wholeDuration;
+      scaledSegments.push({ scaledStart,
+                            scaledEnd,
+                            bufferedInfos });
+    }
+  }
+  return scaledSegments;
+}
+
+/**
+ * Display a graph representing what has been buffered according to the data
+ * given.
+ * Allow to seek on click, display the current time, and display a tooltip
+ * describing the buffered data when hovering represented data.
+ * @param {Object}
+ */
+export default function BufferContentGraph({
+  currentTime, // The time currently played
+  data, // The buffered data for a single type of buffer (e.g. "audio")
+  maximumPosition, // The maximum seekable position
+  minimumPosition, // The minimum seekable position
+  seek, // function allowing to seek in the content
+  type, // The type of buffer (e.g. "audio", "video" or "text")
+}) {
+  const [ randomColors ] = useState(shuffleArray(COLORS));
+  const [ tipVisible, setTipVisible ] = useState(false);
+  const [ tipPosition, setTipPosition ] = useState(0);
+  const [ tipText, setTipText ] = useState("");
+  const containerEl = useRef(null);
+  const canvasEl = useRef(null);
+  const representationsEncountered = useRef([]);
+  const duration = Math.max(maximumPosition - minimumPosition, 0);
+
+  const currentSegmentsScaled = useMemo(() => {
+    return scaleSegments(data, minimumPosition, maximumPosition);
+  }, [data, minimumPosition, maximumPosition]);
+
+  useEffect(() => {
+    if (canvasEl === null || canvasEl === undefined) {
+      return ;
+    }
+    const ctx = canvasEl.current.getContext("2d");
+    if (ctx === null) {
+      return;
+    }
+    canvasEl.current.width = CANVAS_WIDTH;
+    canvasEl.current.height = CANVAS_HEIGHT;
+    clearCanvas(ctx, CANVAS_WIDTH, CANVAS_HEIGHT);
+
+    if (minimumPosition === undefined ||
+        maximumPosition === undefined ||
+        minimumPosition >= maximumPosition)
+    {
+      return;
+    }
+    for (let i = 0; i < currentSegmentsScaled.length; i++) {
+      const segment = currentSegmentsScaled[i];
+      const representation = segment.bufferedInfos.infos.representation;
+      let indexOfRepr = representationsEncountered
+        .current
+        .indexOf(representation);
+      if (indexOfRepr < 0) {
+        representationsEncountered.current.push(representation);
+        indexOfRepr = representationsEncountered.current.length - 1;
+      }
+      const colorIndex = indexOfRepr % COLORS.length;
+      const color = randomColors[colorIndex];
+      const startX = segment.scaledStart * CANVAS_WIDTH;
+      const endX = segment.scaledEnd * CANVAS_WIDTH;
+      ctx.fillStyle = color;
+      ctx.fillRect(Math.ceil(startX),
+                   0,
+                   Math.ceil(endX - startX),
+                   CANVAS_HEIGHT);
+    }
+    if (typeof currentTime === "number" &&
```

Comment:
Can you factorize the "buffers" and "position" parts ?

Reply:
will do

Reply:
done